src/platform_impl/windows/event_loop.rs

```diff
@@ -393,57 +425,84 @@ impl EventLoop {
         //
         runner.prepare_wait();
 
-        let control_flow_timeout = match runner.control_flow() {
-            ControlFlow::Wait => None,
-            ControlFlow::Poll => Some(Duration::ZERO),
-            ControlFlow::WaitUntil(wait_deadline) => {
-                let start = Instant::now();
-                Some(wait_deadline.saturating_duration_since(start))
-            },
-        };
-        let timeout = min_timeout(control_flow_timeout, timeout);
+        {
+            let control_flow_timeout = match runner.control_flow() {
+                ControlFlow::Wait => None,
+                ControlFlow::Poll => Some(Duration::ZERO),
+                ControlFlow::WaitUntil(wait_deadline) => {
+                    let start = Instant::now();
+                    Some(wait_deadline.saturating_duration_since(start))
+                },
+            };
+            timeout = min_timeout(timeout, control_flow_timeout);
+        }
```

Comment:
```suggestion
        let timeout = {
            let control_flow_timeout = match runner.control_flow() {
                ControlFlow::Wait => None,
                ControlFlow::Poll => Some(Duration::ZERO),
                ControlFlow::WaitUntil(wait_deadline) => {
                    let start = Instant::now();
                    Some(wait_deadline.saturating_duration_since(start))
                },
            };
            min_timeout(timeout, control_flow_timeout)
        };
```

Slightly cleaner this way

Reply:
Should I declare new binding `let timeout` or just update existing variable `timeout`? I would update code tomorrow.